src/Raven.Server/Commercial/SetupManager.cs

```diff
@@ -0,0 +1,365 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Net;
+using System.Net.Http;
+using System.Security.Cryptography.X509Certificates;
+using System.Text;
+using System.Threading;
+using System.Threading.Tasks;
+using Certes;
+using Certes.Acme;
+using Certes.Pkcs;
+using Microsoft.AspNetCore.Builder;
+using Microsoft.AspNetCore.Hosting;
+using Microsoft.AspNetCore.Http;
+using Microsoft.Extensions.DependencyInjection;
+using Newtonsoft.Json;
+using Raven.Client.Documents.Operations;
+using Raven.Server.Config;
+using Raven.Server.ServerWide;
+using Raven.Server.Web.Authentication;
+using Sparrow.Platform;
+using Sparrow.Platform.Posix;
+
+namespace Raven.Server.Commercial
+{
+    public class SetupManager : IDisposable
+    {
+        public const string SettingsFileName = "settings.json";
+        public const string RavenDbDomain = "dbs.local.ravendb.net";
+
+        private readonly ServerStore _serverStore;
+        public Timer CertificateRenewalTimer { get; set; }
+
+        public SetupManager(ServerStore serverStore)
+        {
+            _serverStore = serverStore;
+            
+            // TODO If we are the leader (and in lets encrypt setup mode), start the timer with the renew task
+        }
+        
+        public async Task<Uri> LetsEncryptAgreement(string email)
+        {
+            using (var acmeClient = new AcmeClient(WellKnownServers.LetsEncryptStaging))
+            {
+                var account = await acmeClient.NewRegistraton("mailto:" + email);
+                return account.GetTermsOfServiceUri();
+            }
+        }
+
+        public async Task<IOperationResult> FetchCertificateTask(Action<IOperationProgress> onProgress, CancellationToken token, SecuredSetupInfo setupInfo)
```

Comment:
In addition to progress, need to also add logs. We might be interested in seeing what is going on there from the logs after the fact